app/controllers/comments_controller.rb

```diff
@@ -7,6 +7,20 @@ class CommentsController < ApplicationController
 
   def create
     redirect_to @post unless @new_comment.update(comment_params)
+
+    if @new_comment.parent_comment.present?
+      return if @new_comment.parent_comment.user == @new_comment.user
+
+      Notification.create(user: @new_comment.parent_comment.user,
```

Comment:
Code logic to create notification for the related resources repeated many times, i suggested you should write a service for it, in longterm and the scaleable of the app, maybe you will have to put create notification logic in many controllers and also the notification itself maybe the logic will more complicated, so encapsulate it in a service will help it more maintainable and reusable. In that service you should have something like:

`def initialize(user, notify_type, sourceable)
`